src/main/php/com/mongodb/io/Connection.class.php

```diff
@@ -199,6 +202,12 @@ private function read0($n) {
     return $b;
   }
 
+  /** Sets socket timeout */
+  public function timeout(float $timeout): self {
+    $this->socket->setTimeout($timeout);
+    return $this;
+  }
```

Comment:
Should this maybe be soemthing along the lines of the following?

```php
$conn->withTimeout(1, fn() => $conn->send(Connection::OP_MS, /* ... */));
```

Reply:
No, this would make it less reusable; I think the `timeout()` method is just fine.